llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1005,6 +1005,33 @@ void VPlanTransforms::simplifyRecipes(VPlan &Plan, Type &CanonicalIVTy) {
   }
 }
 
+/// Return true if \p Cond is known to be true for given \p BestVF and \p
+/// BestUF.
+static bool isConditionKnown(VPValue *Cond, VPlan &Plan, ElementCount BestVF,
```

Comment:
The condition could be known to be either false or true, so to avoid ambiguity perhaps this should be renamed to `isBranchConditionKnownTrue`? I also think adding the word `Branch` indicates the condition relates to the terminator branch condition, as opposed to the condition for a select, etc.

Reply:
I updated it to `isConditionTrueViaVFAndUF`. I've not included `Branch` for now, as I think the logic is independent of whether the condition is used in a branch or otherwise.